File: day02_1.jl
using DelimitedFiles

data_file = "./data/reduced/day02_input.txt"
# data_file = "./data/full/day02_input.txt"
data = readdlm(data_file, ' ')
C = [Char(d[1]) for d in data[:, 1]]
X = Vector{Int}(data[:, 2])

position, depth = 0, 0

for (com, x) in zip(C, X)
    global position, depth
    if com == 'f'
        position += x
    elseif com == 'u'
        depth -= x
    elseif com == 'd'
        depth += x
    end
end

position_x_depth = position * depth
@show position, depth
@show position_x_depth
answer = position_x_depth

File: day02_1.js
const fs = require("fs");

// data_file = "./data/full/day02_input.txt";
data_file = "data/reduced/day02_input.txt"
const data = fs.readFileSync(data_file, "utf8").split("\n");

const directions = data
  .map((i) => i.split(" "))
  .map((item) => [item[0][0], parseInt(item[1], 10)]);

const state = directions.reduce(
  (state, command) => {
    switch (command[0]) {
      case "f":
        state[0] += command[1];
        break;
      case "u":
        state[1] -= command[1];
        break;
      case "d":
        state[1] += command[1];
        break;
    }
    return state;
  },
  [0, 0]
);

position_x_depth = state.reduce((x, y) => x * y);

console.log("position_x_depth =", position_x_depth);
